lib/levels/rooms_creation.rb

```diff
@@ -0,0 +1,209 @@
+
+     for j in 0..(number_lo-1)
+       l = learning_objects[order[j].to_i]
+       RoomsLearningObject.create(:room_id => room.id,:learning_object_id => l.id)
+     end
+
+      room.id
+    end
+
+    def self.compute_limit(number_lo, learning_objects,setup)
+      weight_solved = 5
+      number_questions = number_lo * 0.6
+
+      sum_dif = 0
+      sum_imp = 0
+
+      for i in 0..(number_lo-1)
+        l = learning_objects[i]
+        lo = LearningObject.where("id = ?", l.id).first
+
+        dif = lo.difficulty.to_s
+        dif_value = 0
+        imp = lo.importance.to_s
+        imp_value = 0
+
+        if dif.to_s == "trivial"
+          dif_value = 0.01
+        elsif dif.to_s == "easy"
+          dif_value = 0.25
+        elsif dif.to_s == "medium"
+          dif_value = 0.5
+        elsif dif.to_s == "hard"
+          dif_value = 0.75
+        elsif dif.to_s == "impossible"
+          dif_value = 1.0
+        else
+          dif_value = 0.5
+        end
+
+        dif_compute = 0.0
+        dif_result = 0.0
+        results = Levels::Preproces.preproces_data(setup)
+
+        all = 0
+        do_not_know_value = 0
+        results.each do |r|
+          if r[0][1].to_i == lo.id
+            all +=1
+            if r[1].to_i == 0
+              do_not_know_value +=1
+            end
+          end
+        end
+
+        if all!= 0
+          dif_compute = do_not_know_value.to_f / all.to_f
+          dif_result = (dif_value + dif_compute) / 2.0
+        else
+          dif_result = dif_value
+        end
+
+        sum_dif +=dif_result
+
+        if imp.to_s == "1"
+          imp_value = 0.0
+        elsif imp.to_s == "2"
+          imp_value = 0.5
+        elsif imp.to_s == "3"
+          imp_value = 1
+        else
+          imp_value = 0.5
+        end
+
+        sum_imp +=imp_value
+
+      end
+
+      avg_dif = sum_dif.to_f / number_lo.to_f
+      avg_imp = sum_imp.to_f / number_lo.to_f
+
+      number_questions * weight_solved * avg_imp * avg_dif
+
+    end
+    def self.prepare_description(number_lo)
+      description = "Najviac zastúpené témy: "
+      number = 5
+      list = Hash.new
+
+      for i in 0..(number_lo-1)
+        l = @sorted_los[i]
+        lo = LearningObject.where("id = ?", l.id).first
+        concepts = lo.concepts
+        concepts.each do |c|
+          if !(list.has_key?(c.name.to_s))
+            list[c.name.to_s] = 1
+          else
+            list[c.name.to_s] = list[c.name.to_s].to_i + 1
+          end
+        end
+      end
+
+      list = list.sort_by{|_key, value| value}.reverse.to_h
+
+      if list.length < number
+        number = list.length.to_i
+      end
+
+
+      for i in 0..(number - 1)
+        if i!= (number-1)
+          description = description + list.keys[i] + " => " + list.values[i].to_s + " krát, "
+        else
+          description = description + list.keys[i] + " => " + list.values[i].to_s + " krát."
```

Comment:
A ruby-way je neist po indexoch, ale pouzivat iteratory (`each`, `map`, ...), ak to treba obmedzit nejakym poctom ako tento kod, tak existuje metoda `take(n)` a netreba ani riesit, ze to ma menej prvkov (ta podmienka `list.length < number` vyssie.

Cize staci:
`description = list.take(number).map{ |key, value| "#{key} => #{value}-krat"}.join(', ') + '.'`

A inak pouzivat string-interpolation miesto skladania cez `+` a netreba davat zbytocne `.to_s`.